src/bun.js/test/expect.zig

```diff
@@ -961,12 +963,30 @@ pub const Expect = struct {
                 }
             }
         } else if (value.isString() and expected.isString()) {
-            const value_string = value.toString(globalObject).toSlice(globalObject, default_allocator).slice();
-            const expected_string = expected.toString(globalObject).toSlice(globalObject, default_allocator).slice();
-            if (strings.contains(value_string, expected_string)) {
-                pass = true;
-            } else if (value_string.len == 0 and expected_string.len == 0) { // edge case two empty strings are true
-                pass = true;
+            if (expected.isStringObjectLike() and value.isStringLiteral()) pass = false else {
+                const value_string = value.toString(globalObject).toSlice(globalObject, default_allocator);
+                defer value_string.deinit();
+                const expected_string = expected.toString(globalObject).toSlice(globalObject, default_allocator);
+                defer expected_string.deinit();
+
+                // jest does not have a `typeof === "string"` check for `toContainEqual`.
+                // it immediately spreads the value into an array.
+
+                var expected_codepoint_cursor = strings.CodepointIterator.Cursor{};
+                var expected_iter = strings.CodepointIterator.init(expected_string.slice());
+                _ = expected_iter.next(&expected_codepoint_cursor);
+
```

Comment:
we're iterating over it character by character after already converting to UTF-8?

can't we use strings.indexOf?

Reply:
good point